app/src/main/java/ch/epfl/sweng/SDP/game/drawing/DrawingOnline.java

```diff
@@ -0,0 +1,165 @@
+package ch.epfl.sweng.SDP.game.drawing;
+
+import android.content.Intent;
+import android.graphics.Typeface;
+import android.os.Bundle;
+import android.os.CountDownTimer;
+import android.support.annotation.NonNull;
+import android.util.Log;
+import android.view.KeyEvent;
+import android.widget.TextView;
+
+import ch.epfl.sweng.SDP.R;
+import ch.epfl.sweng.SDP.auth.Account;
+import ch.epfl.sweng.SDP.firebase.Database;
+import ch.epfl.sweng.SDP.game.VotingPageActivity;
+import ch.epfl.sweng.SDP.home.HomeActivity;
+import ch.epfl.sweng.SDP.localDatabase.LocalDbHandlerForImages;
+import ch.epfl.sweng.SDP.matchmaking.GameStates;
+import ch.epfl.sweng.SDP.matchmaking.Matchmaker;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.ValueEventListener;
+
+public class DrawingOnline extends DrawingOffline {
+
+    private int time;
+    private int timeInterval;
+    private String winningWord;
+
+    private static final String TOP_ROOM_NODE_ID = "realRooms";
+
+    private String roomId;
+
+    private final Database database = Database.INSTANCE;
+    private DatabaseReference timerRef;
+    private DatabaseReference stateRef;
+    private boolean isVotingActivityLaunched = false;
+
+    @Override
+    protected int getLayoutid() {
+        return R.layout.activity_drawing;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        time = 60000; //will be passed as variable in future, not hardcoded
+        timeInterval = 1000;  //will be passed as variable in future, not hardcoded
+
+        Typeface typeMuro = Typeface.createFromAsset(getAssets(), "fonts/Muro.otf");
+        ((TextView) findViewById(R.id.timeRemaining)).setTypeface(typeMuro);
+
+        String path = TOP_ROOM_NODE_ID + "." + roomId + ".timer.observableTime";
+        timerRef = database.getReference(path);
+        timerRef.addValueEventListener(listenerTimer);
+        stateRef = database.getReference(TOP_ROOM_NODE_ID + "." + roomId + ".state");
+        stateRef.addValueEventListener(listenerState);
+
+        Intent intent = getIntent();
+
+        roomId = intent.getStringExtra("RoomID");
+        winningWord = intent.getStringExtra("WinningWord");
+
+        setCountdownTimer();
+
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+
+        // Does not leave the room if the activity is stopped because
+        // voting activity is launched.
+        if (!isVotingActivityLaunched) {
+            Matchmaker.getInstance(Account.getInstance(this)).leaveRoom(roomId);
+        }
```

Comment:
This is useless since isVotingActivityLaunched is always false

Reply:
Sure, I will fix this

Reply:
I fixed this. Can you take a look at my solution?

Reply:
@LudoHoff Ye I added this and I forgot to remove it later on after adding the loading screen